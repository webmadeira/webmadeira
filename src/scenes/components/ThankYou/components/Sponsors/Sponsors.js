import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { mixins } from '../../../../../theme'

const Root = styled.ul`
  margin: 0;
  margin-top: 50px;
  padding: 0;
  display: flex;
  justify-content: center;
  
  ${mixins.mediaQuery.phone`
      flex-direction: column;
      align-items: center;
  `};
`

const Sponsor = styled.li`
  list-style: none;
  padding: 0 25px;
`

const renderSponsors = sponsors => sponsors.map(sponsor => (
  <Sponsor key={sponsor.id}>
    <a href={sponsor.link} target="_blank" rel="noopener noreferrer">
      <img src={sponsor.logo} alt={sponsor.name} />
    </a>
  </Sponsor>
))

const Sponsors = ({ sponsors }) => (
  <Root>
    {renderSponsors(sponsors)}
  </Root>
)

Sponsors.propTypes = {
  sponsors: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
}

export default Sponsors
