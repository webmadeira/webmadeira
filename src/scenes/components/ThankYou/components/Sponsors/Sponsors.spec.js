import React from 'react'
import { shallow } from 'enzyme'
import Sponsors from './Sponsors'

describe('Sponsors component', () => {
  it('must render properly', () => {
    const sponsors = [
      {
        id: 0,
        name: '',
        logo: '',
        link: '',
      },
    ]

    const component = shallow(<Sponsors sponsors={sponsors} />)

    expect(component).toMatchSnapshot()
  })
})
