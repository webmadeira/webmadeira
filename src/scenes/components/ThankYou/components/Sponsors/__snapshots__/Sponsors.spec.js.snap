// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sponsors component must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sponsors
    sponsors={
      Array [
        Object {
          "id": 0,
          "link": "",
          "logo": "",
          "name": "",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.li>
          <a
            href=""
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt=""
              src=""
            />
          </a>
        </styled.li>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "children": <a
            href=""
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt=""
              src=""
            />
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt=""
              src=""
            />,
            "href": "",
            "rel": "noopener noreferrer",
            "target": "_blank",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "src": "",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "a",
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.li>
            <a
              href=""
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt=""
                src=""
              />
            </a>
          </styled.li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "children": <a
              href=""
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt=""
                src=""
              />
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt=""
                src=""
              />,
              "href": "",
              "rel": "noopener noreferrer",
              "target": "_blank",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "src": "",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "a",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
